pkgs/build-support/testers/hasPkgConfigModules/tester.nix

```
@@ -30,18 +29,20 @@ runCommand testName {
         }
         package.meta;
   } ''
-    echo "checking pkg-config module $moduleName in $buildInputs"
-    set +e
-    version="$(pkg-config --modversion $moduleName)"
-    r=$?
-    set -e
-    if [[ $r = 0 ]]; then
-      echo "✅ pkg-config module $moduleName exists and has version $version"
-      echo "$version" > $out
-    else
-      echo "These modules were available in the input propagation closure:"
-      pkg-config --list-all
-      echo "❌ pkg-config module $moduleName was not found"
-      false
-    fi
+    for moduleName in ${lib.escapeShellArgs moduleNames}; do
```

```suggestion
    for moduleName in $moduleNames; do
```

(and define a `moduleNames` variable above)

This way we create less source garbage in the store.

---

`buildCommand` is written as a file? Done.

---

Yeah, it's a `passAsFile`.

---

Yeah, I think it's just written in the build sandbox.